### PR TITLE
[ResourceList] Adjust sizing of CheckableButtonWrapper so it doesn't get squished

### LINKED ISSUE #2961: [ResourceList] Improve styling of `.HeaderContentWrapper` children
## Issue summary

- Basically summarized here: https://github.com/Shopify/polaris-react/pull/2957#discussion_r419640880
- The styling of the `SelectWrapper` could be improved to have a `min-width` of at least the checkbox width + `spacing()`. 

## Expected behavior

- Upon fixing a bug in `ResourceList`'s styling, we noticed an improvement to be made. Adding `@include layout-flex-fix;` to the children of `.HeaderContentWrapper` should result in proper truncation of the children.  

## Actual behavior

- However, in cases where the sortOption label is insanely long, it starts encroaching onto the `checkbox`, resulting in scenarios as such: 
image

## Steps to reproduce the problem

1. Paste this in `ResourceList.scss`: 

```scss
.HeaderTitleWrapper,
.CheckableButtonWrapper,
.AlternateToolWrapper,
.SortWrapper,
.SelectButtonWrapper {
  @include layout-flex-fix;
}
```

<details>
<summary>2. Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useState} from 'react';

import {Avatar, Card, ResourceItem, ResourceList, TextStyle} from '../src';

export function Playground() {
  const [sortValue, setSortValue] = useState('DATE_MODIFIED_DESC');

  const resourceName = {
    singular: 'customer',
    plural: 'customers',
  };

  const items = [
    {
      id: 341,
      url: 'customers/341',
      name: 'Mae Jemison',
      location: 'Decatur, USA',
    },
    {
      id: 256,
      url: 'customers/256',
      name: 'Ellen Ochoa',
      location: 'Los Angeles, USA',
    },
  ];

  return (
    <Card>
      <ResourceList
        selectable
        selectedItems={[]}
        resourceName={resourceName}
        items={items}
        renderItem={renderItem}
        sortValue={sortValue}
        sortOptions={[
          {
            label:
              'Last shrimpy shrimp created (ascending shrimp) shrimp shrimp shrimp shrimp shrimp',
            value: 'DATE_MODIFIED_DESC',
          },
          {label: 'Oldest update', value: 'DATE_MODIFIED_ASC'},
        ]}
        onSortChange={(selected) => {
          setSortValue(selected);
          console.log(`Sort option changed to ${selected}.`);
        }}
      />
    </Card>
  );

  function renderItem(item) {
    const {id, url, name, location} = item;
    const media = <Avatar customer size="medium" name={name} />;

    return (
      <ResourceItem
        id={id}
        url={url}
        media={media}
        accessibilityLabel={`View details for ${name}`}
      >
        <h3>
          <TextStyle variation="strong">{name}</TextStyle>
        </h3>
        <div>{location}</div>
      </ResourceItem>
    );
  }
}


```

</details>

### PR DESCRIPTION
**Pull request by amorriscode**
### WHY are these changes introduced?

Fixes #2961

[Following up on a previous PR](https://github.com/Shopify/polaris-react/pull/2957), the `CheckableButton` was being squished.

### WHAT is this pull request doing?

This PR adds `min-width: spacing();` to the `CheckableButtonWrapper` so it has a nice amount of space.

**Before:**

image

**After:**

image

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React, {useState} from 'react';

import {Avatar, Card, ResourceItem, ResourceList, TextStyle} from '../src';

export function Playground() {
  const [sortValue, setSortValue] = useState('DATE_MODIFIED_DESC');

  const resourceName = {
    singular: 'customer',
    plural: 'customers',
  };

  const items = [
    {
      id: 341,
      url: 'customers/341',
      name: 'Mae Jemison',
      location: 'Decatur, USA',
    },
    {
      id: 256,
      url: 'customers/256',
      name: 'Ellen Ochoa',
      location: 'Los Angeles, USA',
    },
  ];

  return (
    <Card>
      <ResourceList
        selectable
        selectedItems={[]}
        resourceName={resourceName}
        items={items}
        renderItem={renderItem}
        sortValue={sortValue}
        sortOptions={[
          {
            label:
              'Last shrimpy shrimp created (ascending shrimp) shrimp shrimp shrimp shrimp shrimp',
            value: 'DATE_MODIFIED_DESC',
          },
          {label: 'Oldest update', value: 'DATE_MODIFIED_ASC'},
        ]}
        onSortChange={(selected) => {
          setSortValue(selected);
          console.log(`Sort option changed to ${selected}.`);
        }}
      />
    </Card>
  );

  function renderItem(item) {
    const {id, url, name, location} = item;
    const media = <Avatar customer size="medium" name={name} />;

    return (
      <ResourceItem
        id={id}
        url={url}
        media={media}
        accessibilityLabel={`View details for ${name}`}
      >
        <h3>
          <TextStyle variation="strong">{name}</TextStyle>
        </h3>
        <div>{location}</div>
      </ResourceItem>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -17,6 +17,9 @@
 
 ### Bug fixes
 
+- Fixed incorrect `icon` color of `Button` when `destructive` and `plain` ([#2958](https://github.com/Shopify/polaris-react/issues/2958))
+- Fixed sizing issue with `CheckableButtonWrapper` on `ResourceList` ([#2961](https://github.com/Shopify/polaris-react/issues/2961))
```
Comment (alex-page) [inline UNRELEASED.md:21]:
```suggestion
- Fixed sizing issue with `CheckableButtonWrapper` on `ResourceList` ([#2961](https://github.com/Shopify/polaris-react/issues/2961))
```

File: src/components/ResourceList/ResourceList.scss [hunk 2]
```diff
@@ -240,3 +240,16 @@ $item-wrapper-loading-height: rem(64px);
...
   user-select: none;
 }
+
+.HeaderTitleWrapper,
+.CheckableButtonWrapper,
+.AlternateToolWrapper,
+.SortWrapper,
+.SelectButtonWrapper {
+  @include layout-flex-fix;
+}
```
Comment (alex-page) [inline src/components/ResourceList/ResourceList.scss:250]: Are these values needed?

**Comment by kyledurand** [comment]
If this is all in a flex container then I would imagine adding `flex-shrink: 0` to `CheckableButtonWrapper` would have been all that's necessary for this fix. Is that something you tried?

**Comment by alex-page** [comment]
@amorriscode I am going to close this PR but please reopen it if it is still an issue.

**Pull request closed without merging**